pkg/vm/engine/disttae/tools.go

```diff
@@ -1047,3 +1049,141 @@ func genInsertBatch(bat *batch.Batch, m *mpool.MPool) (*api.Batch, error) {
 func genColumnPrimaryKey(tableId uint64, name string) string {
 	return fmt.Sprintf("%v-%v", tableId, name)
 }
+
+func appendBytes(v any, buf *bytes.Buffer) {
+	if v == nil {
+		panic("should not be nil")
+	}
+	switch v := v.(type) {
+	case bool:
+		if v {
+			buf.WriteByte(byte(1))
+		} else {
+			buf.WriteByte(byte(0))
+		}
+	case int8:
+		buf.Write(types.EncodeInt8(&v))
+	case int16:
+		buf.Write(types.EncodeInt16(&v))
+	case int32:
+		buf.Write(types.EncodeInt32(&v))
+	case int64:
+		buf.Write(types.EncodeInt64(&v))
+	case uint8:
+		buf.Write(types.EncodeUint8(&v))
+	case uint16:
+		buf.Write(types.EncodeUint16(&v))
+	case uint32:
+		buf.Write(types.EncodeUint32(&v))
+	case uint64:
+		buf.Write(types.EncodeUint64(&v))
+	case float32:
+		buf.Write(types.EncodeFloat32(&v))
+	case float64:
+		buf.Write(types.EncodeFloat64(&v))
+	case []byte:
+		buf.Write(v)
+	case types.Date:
+		buf.Write(types.EncodeDate(&v))
+	case types.Datetime:
+		buf.Write(types.EncodeDatetime(&v))
+	case types.Timestamp:
+		buf.Write(types.EncodeTimestamp(&v))
+	case types.Decimal64:
+		buf.Write(types.EncodeDecimal64(&v))
+	case types.Decimal128:
+		buf.Write(types.EncodeDecimal128(&v))
+	case types.TS:
+		buf.Write(v[:])
+	case types.Rowid:
+		buf.Write(v[:])
+	case types.Uuid:
+		buf.Write(v[:])
+	case memtable.ID:
+		var id uint64
+
+		id = uint64(v)
+		buf.Write(types.EncodeUint64(&id))
+	default:
+		panic(fmt.Sprintf("unknown type: %T", v))
+	}
+}
+
+func genRow(val *DataValue, cols []string) []any {
```

Comment:
I am a little bit surprised by that you need genRow and appendBytes for this.

Reply:
> I am a little bit surprised by that you need genRow and appendBytes for this.

Because of laziness, reused go's map. go's map does not support arbitrary keys. I'll switch to another map later, but for now, I'm using go's map directly for speed. @fengttt

Reply:
In order to use go's map, I need to convert the structure of YueSheng to string for the key. This will be removed later.But I think go's map is the most robust, so I used go's map first, I was afraid that using other hashmaps would introduce other problems.